src/main/java/ru/svgogin/service/spark/handler/RestResponseEntityExceptionHandler.java

```diff
@@ -1,38 +1,86 @@
 package ru.svgogin.service.spark.handler;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.ConstraintViolationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.lang.NonNull;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 import ru.svgogin.service.spark.errordto.ErrorDto;
+import ru.svgogin.service.spark.errordto.ErrorDto.ErrorCode;
 import ru.svgogin.service.spark.exception.EntityAlreadyExistsException;
 import ru.svgogin.service.spark.exception.NoSuchEntityException;
 
 @ControllerAdvice
 public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
   private static final Logger log = LoggerFactory
       .getLogger(RestResponseEntityExceptionHandler.class);
+  ErrorCode errorCode;
```

Comment:
why do you this?
This class is a singleton and can work in multiple threads. In this case you can return incorrect result

Reply:
Fixed